RIRImageResizing/Classes/NSObject/RIRResizeImageOperationParameters.h

```diff
@@ -0,0 +1,43 @@
+//
+//  RIRResizeImageOperationParameters.h
+//  Pods
+//
+//  Created by Richard Reitzfeld on 11/21/16.
+//
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+
+
+
+
+typedef NS_ENUM(NSInteger, UIImage_RIRResizing_ResizeMode) {
+    UIImage_RIRResizing_ResizeMode_ScaleToFill,
+    UIImage_RIRResizing_ResizeMode_AspectFit,
+    UIImage_RIRResizing_ResizeMode_AspectFill,
+    
+    UIImage_RIRResizing_ResizeMode_first = UIImage_RIRResizing_ResizeMode_ScaleToFill,
+    UIImage_RIRResizing_ResizeMode_last = UIImage_RIRResizing_ResizeMode_AspectFill
+};
+
+
+
+
+@interface RIRResizeImageOperationParameters : NSObject
+
+#pragma mark - newSize
+@property (nonatomic) CGSize newSize;
```

Comment:
All these properties need extra descriptions, like `assign`. Also, all the required parameters that are set init, need to be read only externally, since they shouldn't be able to be set after init.

I'm actually realizing, even though scale isn't required, we need it to also be in the init method, and read only (publicly). This is because, we don't want any confusion by allowing the scale value to be set after the resized image has been generated.